lib/runners/VapidDeployer/index.js

```diff
@@ -0,0 +1,318 @@
+    const hadUploads = await this.uploadFiles(presignedPosts);
+
+    // Update site
+    const manifest = this.generateManifest(siteChecksums, presignedPosts);
+    const site = await this.updateSite(manifest);
+
+    if (hadUploads) {
+      Logger.info('Upload complete');
+    }
+
+    Logger.extra(`View your website at ${site.url}`);
+
+    // Clean up
+    Utils.removeFiles(this.tmpDir);
+
+    return;
+  }
+
+  /*
+   * Checks to see if the user is logged in. If not,
+   * prompts them to log in.
+   */
+  async ensureLogin() {
+    const loggedIn = !Utils.isEmpty(this.credentials);
+
+    if (loggedIn) return;
+
+    Logger.info('Please enter your Vapid credentials, or visit vapid.com to signup.');
+    const answers = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'email',
+        message: 'Email:',
+      },
+
+      {
+        type: 'password',
+        name: 'password',
+        message: 'Password:',
+      },
+    ]);
+
+    const endpoint = url.resolve(this.apiURL, '/auth/me');
+    const body = { email: answers.email, password: answers.password };
+    const response = await got.post(endpoint, { body, json: true });
+    const userNetrc = netrc();
+
+    this.credentials = {
+      login: answers.email,
+      password: response.body.api_key,
+    };
+    
+    userNetrc[this.apiHostname] = this.credentials;
+    return netrc.save(userNetrc);
+  }
+
+  /*
+   * Check to see if site already has a corresponding hosting siteID.
+   * If not, ask them a few questions.
+   */
+  async ensureSite() {
+    const { site } = this.config;
+    let answers;
+
+    if (site) {
+      return site;
+    }
+
+    // Ask questions
+    answers = await inquirer.prompt({
+      type: 'confirm',
+      name: 'newSite',
+      message: 'Is this a new website?',
+    });
+
+    if (!answers.newSite) {
+      Logger.extra([
+        'Please add your site\'s ID to package.json.',
+        // TODO: See vapid.com/blah-blah for more info
+      ]);
+    }
+
+    answers = await inquirer.prompt({
+      type: 'confirm',
+      name: 'deployContent',
+      message: 'Would you like to deploy the existing content too?',
+    });
+
+    // Create a new site
+    const deployContent = answers.deployContent;
+    const endpoint = url.resolve(this.apiURL, '/sites');
+    const response = await got.post(endpoint, { json: true, headers: this.getHeaders() });
+    this.siteId = response.body.site.id;
+    
+    // Update package.json
+    const pjson = require(this.paths.pjson); // eslint-disable-line global-require, import/no-dynamic-require
+    pjson.vapid.site = this.siteId;
+    writePkg.sync(this.paths.pjson, pjson);
+
+    if (deployContent) {
+      Logger.info('Preparing content')
+      this.content = await this.getContent();
+
+      if (fs.existsSync(this.paths.uploads)) {
+        Utils.copyFiles(this.paths.uploads, path.resolve(this.tmpDir, 'uploads'));
+      }
+    }
+
+  return response.body.site.id;
+
+    return site;
```

Comment:
Double return!

Reply:
I like to keep a spare return around. You never know when you'll need one! 😜